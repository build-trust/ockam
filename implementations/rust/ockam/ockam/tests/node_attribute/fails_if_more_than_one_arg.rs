// Test case to verify that only one argument is passed.
//
#[ockam::node]
async fn foo(c: ockam::Context, _x: u64) {
    c.stop().unwrap();
}
