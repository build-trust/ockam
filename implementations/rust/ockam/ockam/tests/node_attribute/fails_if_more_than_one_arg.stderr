error: a function with '#[ockam::node]' must have exactly one argument
 --> $DIR/fails_if_more_than_one_arg.rs:4:7
  |
4 | async fn foo(c: ockam::Context, _x: u64) {
  |       ^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fails_if_more_than_one_arg.rs:3:1
  |
3 | / #[ockam::node]
4 | | async fn foo(c: ockam::Context, _x: u64) {
5 | |     c.stop().unwrap();
6 | | }
  | |_^ consider adding a `main` function to `$DIR/tests/node_attribute/fails_if_more_than_one_arg.rs`
